### PR TITLE
Agn sources

### PR DESCRIPTION
**Pull request by Henry-Best-01**
Functions to calculate response functions of the accretion disk due to X-ray illumination are completed.
Docstrings included for all functions
Testing included for all functions
Quality assurance done by comparing this code with Amoeba, which was previously compared to an independent code written by James H.H. Chan

* As of now all the functions are in astro_util. However, the last function may belong within the AGN class. Please let me know where you think it belongs since this represents the reprocessing that was the goal to be calculated!

### CODE REVIEW

**Comment by Henry-Best-01** [comment]
I'm unsure why my pre-commit is failing, the message is:

- hook id: ruff
- exit code: 2

ruff failed
  Cause: Failed to create cache file '/code/.ruff_cache/0.1.12/14578334182369668236'
  Cause: No such file or directory (os error 2)

**Comment by sibirrer** [comment]
Thank you @Henry-Best-01 ! Yes, we have noticed the pre-commit failure in other recent PRs. I doubt it's your lines of code changes. I suggest we review it and see whether we can fix the pre-commit part separately.

**Comment by sibirrer** [comment]
pre-commit issue being assigned #121

**Comment by ajshajib** [comment]
I have reverted the the `ruff` version to a little older one, which used to work. So, please sync with the latest main by pulling, which will hopefully make things pass. @Henry-Best-01

File: slsim/Util/astro_util.py [hunk 1]
```diff
@@ -191,20 +198,20 @@ def create_radial_map(r_out, r_resolution, inclination_angle):
...
-        will be shape (2 * r_resolution), (2 * r_resolution)
-    :param inclination_angle: the inclination of the plane of the accretion disk with
+    :param r_out: The maximum radial value. For an accretion disk, this can be 10^3 to 10^5.
+    :param r_resolution: The number of points between r = 0 and r = r_out. The final map
+        will be shape (2 * r_resolution), (2 * r_resolution).
+    :param inclination_angle: The inclination of the plane of the accretion disk with
         respect to the observer in degrees.
-    :return: a 2-dimensional array of radial positions of shape ((2 * r_resolution), (2
-        * r_resolution)) in the projected plane of the sky
+    :return: A 2-dimensional array of radial positions of shape ((2 * r_resolution), (2
```
Comment (ajshajib) [inline slsim/Util/astro_util.py:206]:
```suggestion
    :return: A 2D array of radial positions of shape ((2 * r_resolution), (2
```

**Comment by ajshajib** [comment]
The pre-commit issue has been fixed. It would also clear up similar errors if new commits are made in the other PRs.

**Comment by nkhadka21** [comment]
Hi @Henry-Best-01 , thank you for this PR! I have a minor comment on a function otherwise good to merge.

**Review by nkhadka21 — APPROVED** [review]

**Pull request merged**